### PR TITLE
Added card and bottom builders logic to handle a new product

### PR DESCRIPTION
**Pull request by GuilhermeMota93**
## RESOURCES
[The Figma file](https://www.figma.com/file/QAkdm5HZdLlG0b8vUkv1pX/Products---Add-%26-Edit-Products?node-id=460%3A1760&viewport=-1704%2C1577%2C0.5334523320198059) and [the initial issue](https://github.com/woocommerce/woocommerce-android/issues/2707)

## CHANGES / DESCRIPTION

This PR will contain the following:
- **add** product card default list logic
- **add** product bottom details list logic
- the way the cards list and the bottom list are in sync

In this PR I added the logic to match the Figma designs regarding the **cards** and the the **bottom** list details.

Taking the cards part, I opened and extended `ProductDetailCardBuilder` to a new `ProductAddDetailCardBuilder` that will simply get the correct details presented if in the **add** path - this is done at the lazy initialisation level of the `ViewModel`
```
private val cardBuilder by lazy {
    when (navArgs.isAddProduct) {
      true -> ProductAddDetailCardBuilder(this, resources, currencyFormatter, parameters)
      else -> ProductDetailCardBuilder(this, resources, currencyFormatter, parameters)
    }
}
```

For the bottom details part, I extended `ProductDetailBottomSheetBuilder` into a new`ProductAddDetailBottomSheetBuilder` that also follows a similar check as the above cards, in the `ViewModel`. 
```
private val productDetailBottomSheetBuilder by lazy {
    when (navArgs.isAddProduct) {
      true -> ProductAddDetailBottomSheetBuilder(resources)
      else -> ProductDetailBottomSheetBuilder(resources)
    }
}
```

This way, for both steps, the **add** flow UI will present properly 👌 

I also made the `InventoryFragment` to become a `global action` so it could be presented from the bottom sheet options.

**NOTE**
I was looking to add some tests on the view models, to validate the presented cards but I stumbled upon an issue with the `AppPrefs` usage. I couldn't mock it properly during the `setup` of the tests, causing the run to crash. I reverted my changes for now, but this is something I would like to add soon, to make sure the **add** path presents the correct details.

As always, let me know your thoughts on it 🙏 

## TESTING
- Click on the PRODUCT bottom nav bar button
    - Wait for the screen to finish loading
    - Click on the FAB button
    - Notice the product detail for a new product flow with: title, description, price and product type options
    - Click the "Add more details"
    - Notice the inventory, shipping, categories, tags and short description options
- Scenario 1:
    - Click "Inventory"
    - Notice the inventory screen showing
    - Fill in the details and press done
    - Notice the card added in the list and removed from the "Add more details"
- Scenario 2:
    - Click "Shipping"
    - Notice the shipping screen showing
    - Fill in the details and press done
    - Notice the card added in the list and removed from the "Add more details"
- Scenario 3:
    - Click "Categories"
    - Notice the categories screen showing
    - Fill in the details and press done
    - Notice the card added in the list and removed from the "Add more details"
- Scenario 4:
    - Click "Tags"
    - Notice the tags screen showing
    - Fill in the details and press done
    - Notice the card added in the list and removed from the "Add more details"
- Scenario 5:
    - Click "Short description"
    - Notice the description screen showing
    - Fill in the details and press done
    - Notice the card added in the list and removed from the "Add more details"

## SCREENSHOTS

<image src="https://user-images.githubusercontent.com/7058393/91369867-c1f82600-e804-11ea-9cab-2cd1f33fc1fb.gif" width="320"/><image src="https://user-images.githubusercontent.com/7058393/91369878-c7557080-e804-11ea-9636-0010587eca17.gif" width="320"/><image src="https://user-images.githubusercontent.com/7058393/91369879-c8869d80-e804-11ea-9556-9d33710d93ce.gif" width="320"/><image src="https://user-images.githubusercontent.com/7058393/91369880-c91f3400-e804-11ea-9e2c-77a7656c0171.gif" width="320"/><image src="https://user-images.githubusercontent.com/7058393/91369881-c9b7ca80-e804-11ea-8c2f-baa894463e81.gif" width="320"/>

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/StartProductAddTypeBottomSheetBuilder.kt [hunk 1]
```diff
@@ -9,7 +9,7 @@ import com.woocommerce.android.ui.products.ProductType.SIMPLE
 import com.woocommerce.android.ui.products.ProductType.VARIABLE
 import com.woocommerce.android.ui.products.ProductTypesBottomSheetViewModel.ProductTypesBottomSheetUiItem
 
-class ProductAddTypeBottomSheetBuilder : ProductTypeBottomSheetBuilder {
+class StartProductAddTypeBottomSheetBuilder : ProductTypeBottomSheetBuilder {
```
Comment (GuilhermeMota93) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/StartProductAddTypeBottomSheetBuilder.kt:12]: Renamed this builder to differentiate the opening of the bottom being from either:
- bottom list builder to start add product
- bottom list builder to show from details

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductAddDetailCardBuilder.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package com.woocommerce.android.ui.products
+
+import com.woocommerce.android.R
+import com.woocommerce.android.model.Product
+import com.woocommerce.android.ui.products.models.ProductProperty
+import com.woocommerce.android.ui.products.models.SiteParameters
+import com.woocommerce.android.util.CurrencyFormatter
+import com.woocommerce.android.viewmodel.ResourceProvider
+
+class ProductAddDetailCardBuilder(
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductAddDetailCardBuilder.kt:10]: I'm wondering it's necessary to have a separate card builder for the **add** workflow. Every new product will have `In Stock` stock status by default and you can't really remove it from a product, so perhaps we could just always show it.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductAddDetailCardBuilder.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    viewModel: ProductDetailViewModel,
+    resources: ResourceProvider,
+    currencyFormatter: CurrencyFormatter,
+    parameters: SiteParameters
+) : ProductDetailCardBuilder(viewModel, resources, currencyFormatter, parameters) {
+    override fun getProductTitlePlaceholder(): Int =
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductAddDetailCardBuilder.kt:16]: "Enter title" could be the default placeholder both for **add** & **edit** flow.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductAddDetailBottomSheetBuilder.kt [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package com.woocommerce.android.ui.products
+
+import com.woocommerce.android.analytics.AnalyticsTracker.Stat
+import com.woocommerce.android.model.Product
+import com.woocommerce.android.ui.products.ProductInventoryViewModel.InventoryData
+import com.woocommerce.android.ui.products.ProductNavigationTarget.ViewProductInventory
+import com.woocommerce.android.viewmodel.ResourceProvider
+
+class ProductAddDetailBottomSheetBuilder
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductAddDetailBottomSheetBuilder.kt:9]: I'm not sure we need to have a separate bottom sheet builder. See my comments in `ProductAddDetailCardBuilder`.

**Review by 0nko — COMMENTED** [review]
I've left a couple of notes in the code.

**Comment by GuilhermeMota93** [comment]
Closing this PR as we're keeping parity with the **edit** flow and these are no longer needed

**Pull request closed without merging**